### DISCUSSION TITLE
Linear algebra support?

### DISCUSSION THREAD

**Discussion by ivanstepanovftw** [microsoft/onnxruntime#15209] [Other Q&A]
Hello, community! I would like to know if there is a linear algebra functions/library, like [`torch.unsqueeze`](https://pytorch.org/docs/stable/generated/torch.unsqueeze.html), [`torch.matmul`](https://pytorch.org/docs/stable/generated/torch.matmul.html), [`torch.Tensor.expand`](https://pytorch.org/docs/stable/generated/torch.Tensor.expand.html) and [`torch.sum`](https://pytorch.org/docs/stable/generated/torch.sum.html) for ONNX Runtime for Web?
It is needed to do some post-processing steps for sentence transformer encoder models.

**Comment by ivanstepanovftw** [comment]
I saw [documentation for ONNX Runtime JavaScript API](https://onnxruntime.ai/docs/api/js/interfaces/Tensor-1.html), there is no such methods for Tensor interface.

**Comment by justinchuby** [comment]
I would suggest creating an ONNX model that includes the post processing steps then run it with onnxruntime. Is that something possible for you?
